docs/src/model_creation/events_and_callbacks.md

```diff
@@ -0,0 +1,497 @@
+affect2!(integrator) = (integrator[:X] += 2.0)
+cb2 = DiscreteCallback(condition2, affect2!)
+
+cbs = CallbackSet(cb1, cb2)
+sol = solve(oprob; callback = cbs)
+plot(sol)
+```
+
+### [Simulation termination and other callback-specific features](@id events_model_callbacks_special)
+Callbacks give the user much more control over the affect and condition functions than events. They can e.g. be used to:
+- Print a statement with information of the simulation state under certain conditions (using [`println`](https://docs.julialang.org/en/v1/base/io-network/#Base.println)).
+- Reset a system's state to its value of the previous time step (using the `integrator.uprev` field).
+- Terminate the simulation once some condition is reached (using the [`terminate!` function](https://docs.sciml.ai/DiffEqDocs/stable/features/callback_functions/#Example-2:-Terminating-an-Integration)).
+
+Here we simulate our decay model, but terminates the solution once $X < 1.0$:
+```@example events_callbacks
+condition(u, t, integrator) = (integrator[:X] - 1.0)
+affect!(integrator) = terminate!(integrator)
+cb = ContinuousCallback(condition, affect!)
+sol = solve(oprob; callback = cb)
+plot(sol)
+```
+
+The DiffEqCallbacks package implements a range of ready-made callbacks that are available for the user. A full list of these can be found [here](https://docs.sciml.ai/DiffEqCallbacks/stable/).
+
+## [Additional consideration when using events (or callbacks)](@id events_additional_considerations)
+
+### [Events affecting changing model parameter values](@id events_additional_considerations_parameters)
+It is not uncommon to have events (or callbacks) which change a system parameter. When doing so one has to be careful, since the event (or callback) will also update the original `ODEProblem`. E.g. in the following [birth-death process model](@ref ref), an event updates `p`'s value at time `5.0`:
+```@example events_considerations
+using Catalyst, OrdinaryDiffEq, Plots # hide
+bd_model = @reaction_network begin
+  @discrete_events begin
+    [5.0] => [p ~ 2.0]
+  end
+  (p,d), 0 <--> X
+end
+
+u0 = [:X => 0.1]
+tspan = (0.0, 10.0)
+ps = [:p => 0.2, :d => 0.1]
+
+oprob = ODEProblem(bd_model, u0, tspan, ps)
+sol = solve(oprob)
+plot(sol)
+```
+If we check the original `ODEProblem`, we can confirm that `ps`'s value has been updated:
+```@example events_considerations
+oprob.ps[:p]
+```
+Indeed, if we simulate our model again, the result is different:
+```@example events_considerations
+sol = solve(oprob)
+plot(sol)
+```
+
+To deal with these situations, it is recommended to not pass your `ODEProblem` directly into `solve`, but rather a [`deepcopy`](https://docs.julialang.org/en/v1/base/base/#Base.deepcopy) of it:
+```@example events_considerations
+oprob = ODEProblem(bd_model, u0, tspan, ps)
+sol = solve(deepcopy(oprob), Tsit5())
+nothing # hide
+```
+
+### [Callbacks during jump simulations](@id events_additional_considerations_jump_parameters)
```

Comment:
This is handled properly for events, right? Not that the user can do anything if it isn't, but still.